### PR TITLE
Added pre-commit hook to enforce lint style

### PR DESCRIPTION
**Pull request by bhamodi**
@jordangarcia - accomplished this using the `grunt githooks` module.

### CODE REVIEW

File: grunt/githooks.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  all: {
+    'pre-commit': 'eslint',
```
Comment (jordangarcia) [inline grunt/githooks.js]: this needs to be `grunt eslint` since using just `eslint` requires eslint to be globally installed on the users machine.

**Comment by jordangarcia** [comment]
I checked out this branch and can't commit since there are style failures.  This githook should only be created once its clear to do a merge.  We wouldn't want any contributor unable to commit becuase of existing style errors.

**Comment by bhamodi** [comment]
Yup, just saw that. There's 10 lint warnings left to address. I'll send a PR for that before this gets merged in.

**Comment by bhamodi** [comment]
By the way, it does allow you to commit if there are only lint warnings and no lint errors. Not sure why you weren't able to send a commit? There's only 10 lint warnings, no errors.

**Comment by jordangarcia** [comment]
ah i must have not noticed that, since i wasn't commiting any changed files.

**Comment by bhamodi** [comment]
It's all good. Either way, I'm working on sending a PR for the last 10 warnings. By the way, on the topic of warnings and errors - we might want to revise what we consider to be an error/warning. I say this because only errors will actually stop a commit from happening and force a user to fix the styling. A warning will only display the warnings but still allow the commit through. So if there's things for sure that we don't want committed, we'll have to change the value to `2` in `.eslintrc`.

**Comment by jordangarcia** [comment]
hmm i dont really like making everything errors, usually warnings are style concerns and errors are actual errors that will affect the runtime.

Is there a way to make `grunt eslint` fail on warning.

The way githooks work, is if the pre-commit process returns a non-zero exit code then it wont let you commit.  So it seems like we need a way to have `grunt eslint` return the appropriate exit code to prevent commiting.

**Comment by bhamodi** [comment]
Ya, I looked into making it fail on warnings but the issue is that the `grunt eslint` command returns an exit code of 0 when only warnings exist. I'm not sure how to override the eslint module to return a non-zeo exit code.

**Comment by bhamodi** [comment]
Ok @jordangarcia - just sent in #81 to resolve the last 10 lint warnings.

**Comment by jordangarcia** [comment]
this lgtm

**Pull request merged**